lib/husky_musher/app.py

```diff
@@ -22,6 +22,12 @@ def set_cache_control(response):
 def page_not_found(error):
     return render_template('page_not_found.html'), 404
 
+@app.errorhandler(BadRequest)
+def handle_bad_request(error):
+    message, netid = error.description
+    app.logger.error(f'Bad request: {message}', exc_info=message)
+    return render_template('invalid_netid.html', netid=netid), 400
```

Comment:
Fine for now, but if we ever emit other BadRequests (or if Flask does implicitly for us), then this error handler will break because it assumes the thing raising the error was the NetID check.

Note also that `exc_info` takes an Exception object or exception tuple (as returned by `sys.exc_info()`), otherwise it calls `sys.exc_info()` itself if the value passed is truthy. So I this works as written, but is a little weird to be passing a string.

Reply:
Yes, true that we've essentially repurposed `BadRequest` to be solely for invalid NetIDs here. It may need to be changed down the road, but I thought it was a nicer approach to return the `invalid_netid.html` template in the error handler rather than within the main endpoint so that we can raise an Exception and return a proper error code.

I did have to do a little finagling with the error description because I wanted to pass `netid` to the template without logging PII to the server. I tried looking at the [Flask API docs](https://flask.palletsprojects.com/en/1.1.x/api/) for `exc_info` but I couldn't find anything useful.

Reply:
Ah I see what you're saying now. It's returned by `sys.exc_info()`, as in the [standard sys library](https://docs.python.org/3/library/sys.html).

Reply:
@tsibley I pushed a new commit that handles custom `InvalidNetId` errors instead of the general `BadRequest` exception. I manually redacted netid within the error handler which allows me to pass the entire error object to `exc_info()`. Will you take a look when you get a chance?